failed to resolve: use of undeclared crate or module `tracing`

I'm getting this error:

```
error[E0433]: failed to resolve: use of undeclared crate or module `tracing`
   --> C:\Users\fred\.cargo\registry\src\index.crates.io-6f17d22bba15001f\axum-extra-0.9.4\src\typed_header.rs:161:9
    |
161 |         axum_core::__log_rejection!(rejection_type = Self, body_text = body, status = status,);
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ use of undeclared crate or module `tracing`
    |
    = note: this error originates in the macro `axum_core::__log_rejection` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Why?

--- a/axum-core/src/lib.rs
+++ b/axum-core/src/lib.rs
@@ -50,6 +50,11 @@
 
 #[macro_use]
 pub(crate) mod macros;
+#[doc(hidden)] // macro helpers
+pub mod __private {
+    #[cfg(feature = "tracing")]
+    pub use tracing;
+}
 
 mod error;
 mod ext_traits;

--- a/axum-core/src/macros.rs
+++ b/axum-core/src/macros.rs
@@ -9,12 +9,12 @@ macro_rules! __log_rejection {
         status = $status:expr,
     ) => {
         {
-            tracing::event!(
+            $crate::__private::tracing::event!(
                 target: "axum::rejection",
-                tracing::Level::TRACE,
+                $crate::__private::tracing::Level::TRACE,
                 status = $status.as_u16(),
                 body = $body_text,
-                rejection_type = std::any::type_name::<$ty>(),
+                rejection_type = ::std::any::type_name::<$ty>(),
                 "rejecting request",
             );
         }

--- a/axum-extra/Cargo.toml
+++ b/axum-extra/Cargo.toml
@@ -34,7 +34,7 @@ json-lines = [
 multipart = ["dep:multer"]
 protobuf = ["dep:prost"]
 query = ["dep:serde_html_form"]
-tracing = ["dep:tracing", "axum-core/tracing", "axum/tracing"]
+tracing = ["axum-core/tracing", "axum/tracing"]
 typed-header = ["dep:headers"]
 typed-routing = ["dep:axum-macros", "dep:percent-encoding", "dep:serde_html_form", "dep:form_urlencoded"]
 

